lib/rules/var-spacing.rule.js

```diff
@@ -10,11 +10,77 @@
 
 'use strict';
 
+function validatePunctuators(current, next, context, node) {
+  next.forEach(function (nextPunctuator) {
+    if (current.loc.start.column !== nextPunctuator.loc.start.column) {
+      context.report({
+        message: 'Invalid indent!',
+        node:    node
+      });
+    }
+  });
+}
+
+function findNextPunctuators(node, source, lines) {
+  var after       = source.getTokenAfter(node),
+      nextLine    = lines[node.loc.start.line],
+      punctuators = [];
+
+  // in this case one big var assignment is declared:
+  //
+  // var var1 = {},
+  //     foo  = [];
+  // let blah = '';
+  if (node.declarations.length > 1) {
+    var nodeStartLine = node.loc.start.line;
+    punctuators = node.declarations.filter(function (subToken) {
+      return subToken.init !== null;
+    }).map(function (subToken) {
+      return source.getTokensBetween(subToken.id, subToken.init)[0];
+    }).filter(function (token, index) {
+      return token.type === 'Punctuator' && token.value === '=' && index > 0 && nodeStartLine < token.loc.start.line;
```

Comment:
the condition must be refactored. Especially parts like the ``token`` validation is duplicated logic